engine/gui/widgets.py

```diff
@@ -89,10 +91,83 @@ def create(
                 None,
             )
 
-        return GUISpec(
+        gui_spec = GUISpec(
             assets=[Asset(**a) for a in spec.get("assets", [])],
             buttons=buttons,
             images=[Image(**b) for b in spec.get("images", [])],
             cancel_action=cancel_action,
             initial_selected_button=selected_button,
         )
+
+        cls.validate(gui_spec)
+        return gui_spec
+
+    def validate(self) -> None:
+        """Validates the GUISpec."""
+
+        if not self.buttons and not self.images:
+            raise ValidationError("Must have at least one button or image")
+
+        asset_names = collections.Counter([a.name for a in self.assets])
+        duplicate_assets = [asset for asset, count in asset_names.items() if count > 1]
+        if duplicate_assets:
+            raise ValidationError(f"Duplicate asset name: {duplicate_assets}.")
+
+        button_names = collections.Counter([b.name for b in self.buttons])
+        duplicate_buttons = [name for name, count in button_names.items() if count > 1]
+        if duplicate_buttons:
+            raise ValidationError(f"Duplicate button name: {duplicate_buttons}")
+
+        image_names = collections.Counter([i.image_asset for i in self.images])
+        duplicate_images = [image for image, count in image_names.items() if count > 1]
+        if duplicate_images:
+            raise ValidationError(f"Duplicate image name(s): {duplicate_images}")
+
+        for b_names in button_names.items():
+            if b_names[0] not in list(asset_names.elements()):
+                raise ValidationError(f"Button {b_names[0]} not in assets")
+
+        for i_names in image_names.items():
+            if i_names[0] not in list(asset_names.elements()):
+                raise ValidationError(f"Image {i_names[0]} not in assets")
+
+        # for name in asset_names.elements():
+        #     if name not in [b for b in button_names.elements()]:
+        #         raise ValidationError(f"Asset {name} not in buttons")
+        #     if name not in [i for i in self.images]:
+        #         raise ValidationError(f"Asset {name} not in images")
```

Comment:
Remove commented out code; usually you don't keep that around unless you have a good reason for it.